src/main/java/org/isf/lab/rest/LaboratoryController.java

```diff
@@ -0,0 +1,242 @@
+package org.isf.lab.rest;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.Authorization;
+import org.isf.exa.manager.ExamBrowsingManager;
+import org.isf.exa.model.Exam;
+import org.isf.lab.dto.LaboratoryDTO;
+import org.isf.lab.dto.LaboratoryForPrintDTO;
+import org.isf.lab.dto.LaboratoryRowDTO;
+import org.isf.lab.manager.LabManager;
+import org.isf.lab.mapper.LaboratoryMapper;
+import org.isf.lab.mapper.LaboratoryRowMapper;
+import org.isf.lab.model.Laboratory;
+import org.isf.lab.model.LaboratoryForPrint;
+import org.isf.lab.model.LaboratoryRow;
+import org.isf.patient.manager.PatientBrowserManager;
+import org.isf.patient.model.Patient;
+import org.isf.shared.exceptions.OHAPIException;
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.model.OHExceptionMessage;
+import org.isf.utils.exception.model.OHSeverityLevel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.List;
+
+@RestController
+@Api(value = "/laboratories", produces = MediaType.APPLICATION_JSON_VALUE, authorizations = {@Authorization(value = "basicAuth")})
+public class LaboratoryController {
+
+    private final Logger logger = LoggerFactory.getLogger(LaboratoryController.class);
+
+    @Autowired
+    protected LabManager laboratoryManager;
+    @Autowired
+    protected ExamBrowsingManager examManager;
+    @Autowired
+    private PatientBrowserManager patientBrowserManager;
+    @Autowired
+    private LaboratoryMapper laboratoryMapper;
+    @Autowired
+    private LaboratoryRowMapper laboratoryRowMapper;
+    @Autowired
+    private org.isf.lab.mapper.LaboratoryForPrintMapper laboratoryForPrintMapper;
+
+    public LaboratoryController(LabManager laboratoryManager, PatientBrowserManager patientBrowserManager, ExamBrowsingManager examManager, LaboratoryMapper laboratoryMapper, LaboratoryRowMapper laboratoryRowMapper) {
+        this.laboratoryManager = laboratoryManager;
+        this.patientBrowserManager = patientBrowserManager;
+        this.examManager = examManager;
+        this.laboratoryMapper = laboratoryMapper;
+        this.laboratoryRowMapper = laboratoryRowMapper;
+    }
+
+    @PostMapping(value = "/laboratories", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Integer> newLaboratory(@RequestBody LaboratoryDTO laboratoryDTO, @RequestBody(required = false) List<String> labRow) throws OHServiceException {
```

Comment:
There is quite a bit of discussion on the Internet about using multiple @RequestBody in parameter has the same method. It is not recommended to do so.
What you need to do:
- Create an object that includes the laboratoryDTO and labRow objects,
e.g.
`class LabAndRowsDTO{
    LaboratoryDTO laboratoryDTO;
    List<String> labRow;
}`
- Use this object in the _newLaboratory_ method and access its properties to extract the required information.
`public ResponseEntity<Integer> newLaboratory(@RequestBody LabAndRowsDTO laboratoryAndRowsDTO) OHServiceException throws {...}`

Reply:
I agree with you and updated pr with your advice.